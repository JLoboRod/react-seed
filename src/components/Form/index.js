import React from "react";

import Box from "../Box";

class Form extends React.Component {
  state = {};

  handleChange = ({ target: { name, value } }) => {
    this.setState({ [name]: value });
  }

  render() {
    return (
      <Box>
        <Box>
          <form>
            <input
              type="text"
              name="name"
              className="input"
              value={this.state.name}
              onChange={this.handleChange}
            />
          </form>
        </Box>
        <Box>
          Hello
        </Box>
      </Box>
    );
  }
}

export default Form;
