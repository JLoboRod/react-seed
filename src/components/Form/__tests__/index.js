import React from "react";
import { shallow } from "enzyme";

import Form from "../index";
import Box from "../../Box/index";

describe("<Form />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Form />);

    expect(wrapper).toMatchSnapshot();
  })

  it("handles changes", () => {
    const wrapper = shallow(<Form />);
    const name = "name";
    const value = "Juan";
    const event = {
      target: {
        name,
        value
      }
    };

    wrapper.find(".input").simulate("change", event);

    expect(wrapper.find(".input").prop("name")).toBe(name);
    expect(wrapper.find(".input").prop("value")).toBe(value);
  })
});