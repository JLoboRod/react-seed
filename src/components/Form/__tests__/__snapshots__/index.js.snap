// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders correctly 1`] = `
<Box>
  <Box>
    <form>
      <input
        className="input"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </form>
  </Box>
  <Box>
    Hello
  </Box>
</Box>
`;
